apis/kubedb/v1alpha2/solr_helpers.go

```diff
@@ -262,12 +269,6 @@ func (s *Solr) SetDefaults() {
 		s.Spec.StorageType = StorageTypeDurable
 	}
 
-	if s.Spec.AuthSecret == nil {
-		s.Spec.AuthSecret = &v1.LocalObjectReference{
-			Name: s.SolrSecretName("admin-cred"),
-		}
```

Comment:
We dont need the authSecret defaulting ?

Reply:
no
a method is added to retrieve the name of auth secret.